lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -52,9 +53,28 @@ class tethys::ScienceSensorsSystemPrivate
   //////////////////////////////////
   // Functions for data manipulation
 
+  /// \brief Called when plugin is asked to reload file.
+  /// \param[in] _filepath Path to file to reload.
+  public: void OnReloadData(const ignition::msgs::StringMsg &_filepath)
+  {
+    igndbg << "Reloading file " << _filepath.data() << "\n";
+
+    // Trigger reload and reread data
+    std::lock_guard<std::mutex> lock(this->dataMutex);
+    this->newDataAvailable = true;
+    this->dataPath = _filepath.data();
+  }
+
+  /// \brief mutex for updating data.
+  public: std::mutex dataMutex;
```

Comment:
@arjo129 hmm, isn't this mutex redundant with `mtx`?

Reply:
No mtx only protects reading. This protects the filename as well as whether data is ready. We could elminate `mtx` since it does not really server any purpose at this point.